quantities.benchmark/CreateQuantities.cs

```diff
@@ -1,64 +1,71 @@
+    public DummyStruct(in Double value) => this.value = value;
     public static implicit operator Double(DummyStruct obj) => obj.value;
 }
 
 [MemoryDiagnoser]
+//[EventPipeProfiler(EventPipeProfile.CpuSampling)]
 public class CreateQuantities
 {
     private static readonly Random random = new();
-    private Double value = random.NextDouble();
-
+    private readonly Double value = random.NextDouble();
 
     [Benchmark(Baseline = true)]
-    public DummyObject CreateObject() => new DummyObject(this.value);
+    public DummyObject CreateObject() => new(in this.value);
+
+    [Benchmark]
+    [MethodImpl(MethodImplOptions.NoOptimization)]
+    public DummyStruct CreateStruct() => new(in this.value);
 
     [Benchmark]
-    public DummyObject CreateObjectIn() => new DummyObject(in this.value);
+    public Length CreateFastQuantity() => Length.Of(in this.value).Si<Kilo, Metre>();
 
     [Benchmark]
-    public DummyStruct CreateStruct() => new DummyStruct(this.value);
+    [MethodImpl(MethodImplOptions.NoOptimization)]
+    public Length CreateFastQuantityWithoutOpt() => Length.Of(in this.value).Si<Centi, Metre>();
 
     [Benchmark]
-    public Length CreateLength() => Length.Si<Kilo, Metre>(in this.value);
+    public Data CreateSlowQuantity() => Data.Of(in this.value).Metric<Kilo, Bytes>();
 
     [Benchmark]
-    public Velocity CreateVelocity() => Velocity.Si<Kilo, Metre>(in this.value).Per<Hour>();
+    public Double CreateAllocatingQuantity() => Velocity.Of(in this.value).Si<Kilo, Metre>().Per.Metric<Hour>();
 }
 
 /*
 // * Summary *
 
-BenchmarkDotNet=v0.12.1, OS=arch 
+BenchmarkDotNet=v0.13.5, OS=arch 
 Intel Core i7-8565U CPU 1.80GHz (Whiskey Lake), 1 CPU, 8 logical and 4 physical cores
-.NET Core SDK=7.0.100
-  [Host]     : .NET Core 7.0.0 (CoreCLR 7.0.22.56001, CoreFX 7.0.22.56001), X64 RyuJIT
-  DefaultJob : .NET Core 7.0.0 (CoreCLR 7.0.22.56001, CoreFX 7.0.22.56001), X64 RyuJIT
-
+.NET SDK=7.0.103
+  [Host]     : .NET 7.0.3 (7.0.323.12801), X64 RyuJIT AVX2
+  DefaultJob : .NET 7.0.3 (7.0.323.12801), X64 RyuJIT AVX2
 
-|         Method |       Mean |     Error |    StdDev | Ratio | RatioSD |  Gen 0 | Gen 1 | Gen 2 | Allocated |
-|--------------- |-----------:|----------:|----------:|------:|--------:|-------:|------:|------:|----------:|
-|   CreateObject |  6.6147 ns | 0.0908 ns | 0.0805 ns | 1.000 |    0.00 | 0.0057 |     - |     - |      24 B |
-| CreateObjectIn |  6.0880 ns | 0.0495 ns | 0.0463 ns | 0.921 |    0.01 | 0.0057 |     - |     - |      24 B |
-|   CreateStruct |  0.0547 ns | 0.0147 ns | 0.0138 ns | 0.008 |    0.00 |      - |     - |     - |         - |
-|   CreateLength |  4.2311 ns | 0.0568 ns | 0.0504 ns | 0.640 |    0.01 |      - |     - |     - |         - |
-| CreateVelocity | 10.0543 ns | 0.0173 ns | 0.0135 ns | 1.522 |    0.02 | 0.0057 |     - |     - |      24 B |
+|                       Method |      Mean |     Error |    StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
```

Comment:
Notice that creation has remained constant for simple types such as ´Length´. But creating more complex quantities, such as `Velocity` is now twice as slow and still allocates :-(

Reply:
You are absolutely right. We're already fast. I did set it as a goal for these types to be blazingly fast as a challenge and as I see them as a drop in for `double`.
Let's take this into more balanced consideration as the project evolves.